pylint/checkers/base/basic_checker.py

```diff
@@ -370,14 +370,15 @@ def _check_using_constant_test(
                 call_inferred = None
             if call_inferred:
                 try:
-                    for inf_call in call_inferred:
-                        if inf_call != astroid.Uninferable:
-                            self.add_message(
-                                "missing-parentheses-for-call-in-test",
-                                node=test,
-                                confidence=INFERENCE,
-                            )
-                            break
+                    # Just forcing the generator to infer all elements.
+                    # astroid.exceptions.InferenceError are false positives
+                    # see https://github.com/PyCQA/pylint/pull/8185
+                    list(call_inferred)
```

Comment:
This is a nit, but can't we do the unpacking above where we are already catching `InferenceError`?

Reply:
Sure, that not a nit that's an optimization 😄